tests/auth_rust/tests/test_auth_demo.rs

```diff
@@ -34,8 +40,12 @@ fn unit_test_with_type(t: AlgorithmType, incorrect_pubkey: bool) -> Result<u64,
     unit_test_with_auth(auth_builder(t).unwrap(), incorrect_pubkey)
 }
 
+fn unit_test_get_run_type() -> EntryCategoryType {
```

Comment:
how do we test it with `exec`? 
I think we need to run one test case with `dynamic linking`, while with `exec` too.